[DRAFT] Actions/action_types list with new platform integration.

YulNaumenko opened the pull request:
## Summary

Current PR is a very draft of the base alerting actions UI structure. The goal of this PR is to build the plugin architecture which will be organized according to the new platform Plugin description.

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios
- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

YulNaumenko committed:
[DRAFT] Actions/action_types list with new platform integration.

YulNaumenko committed:
[DRAFT] Actions/action_types list with new platform integration.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/2702/)
* Commit: 585fc2b26b2e3734280714e9e6ae385dba6e7a42

Commit:
Standardizing IconField implementation across the app (#47196)

* Moving FieldNameIcon implementation to FieldIcon implementation in kibana_react directory

* Adding LensFieldIcon implementation around the new FieldIcon to be used in Lens components

* Applying new LensFieldIcon in the Lens components

* Applying new FieldIcon component in Graph components

* Applying new FieldIcon component in the rest of the components

* Adding missing type to lens field icon

* adding missing type

* Addressing PR comments

dmlemeshko committed:
[functional/services] update webdriver lib and types (#47381)

* [functional/services] refactor using new types

* [code/history] wrap forward navigation with browser service

Commit:
Adding width and height as parameters to image url (#46917)

* Adding width and height as parameters to image url

* Addressing PR comments

* Removing unnecessary error message

pheyos committed:
[ML] Add tests to create saved search based jobs (#47323)

This PR adds tests for ML job creation based on saved searches

legrego committed:
Honor current search criteria when exporting saved objects (#47223)

* honor current search criteria when exporting saved objects

* adding core docs

mshustov committed:
To recommend how to access API not available in New platform (#46682)

* To recommend how to access API not available in New platform

* add a note about usage on the client side

* Update src/core/MIGRATION.md

Co-Authored-By: Josh Dover <me@joshdover.com>

smith committed:
[APM] Use new platform for toast notifications (#47276)

* [APM] Use new platform for toast notifications

* fix more tests

* remove comment

joshdover committed:
Make Core interfaces exposed to LP plugins consistent with NP… (#47074)

dmlemeshko committed:
skip flaky test (#46977) (#47437)

Commit:
disable flaky suite (#47184)

streamich committed:
Improve Interpreter TypeScript typings (#47212)

* fix: 🐛 correct createError() implementation and typings

* feat: 🎸 improve expressions service TypeScript types

* perf: ⚡️ improve ExpressionLoader class

* feat: 🎸 improve Interpreter types

* feat: 🎸 improve Interpreter function typings

* feat: 🎸 improve Interepreter test TypeScript typings

* feat: 🎸 TypeScriptify kibana_context expression function

* feat: 🎸 TypeScriptify "kibana" expressions functions

* feat: 🎸 TypeScriptify "visualization" function

* feat: 🎸 TypeScriptify Interpreter initialization

* test: 💍 fix interpreter tests

* feat: 🎸 add TypeScript typing to some of the executor internals

* fix: 🐛 this fixes visualizations; why?

* chore: 🤖 delete redundant files and fix TypeScript error

* perf: ⚡️ use not null assertion for .dataHandler

* refactor: 💡 stricten AnyExpressionFunction type

* fix: 🐛 set correct initial pipeline exec context

* feat: 🎸 improve expression type typings

* test: 💍 call interpreter with correct input from func tests

joshdover committed:
Add types to Elasticsearch API calls (#46668)

Commit:
disable flaky suite (#47312)

YulNaumenko committed:
Added missing requests execution stuff by using HttpServiceBase

YulNaumenko committed:
Added missing requests execution stuff by using HttpServiceBase

FrankHassanabad committed:
[SIEM] [Detection Engine] Adds a create signal API (#47384)

## Summary

* https://github.com/elastic/kibana/issues/47013

This adds a basic create signal API rout API at:

```
/api/siem/signals
```

At the moment this will create an action/alert as a one for one.

This also introduces several scripts which allow developers to:
  * Re-create signals index for development velocity
  * post signals to the `/api/siem/signals` endpoint
  * Check different indexes, saved objects, and task manager for trouble shooting 

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

~~- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~

~~- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~

~~- [ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~

~~- [ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~

~~- [ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

~~- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

~~- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/2872/)
* Commit: c8ea1b8dc98c21c8e814bb45dd966d3d9e23d621

KOTungseth committed:
[DOCS] Removed plugin-dir (#47488)

KOTungseth committed:
[DOCS] Changed setting to deprecated (#47372)

YulNaumenko committed:
Merge branch 'action/action-types-list' of github.com:YulNaumenko/kibana into action/action-types-list

YulNaumenko committed:
Fixed failing build errors

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/2913/)
* Commit: ff3f8186657cc2dd9c01d719a0810a8aab190d38
